pmlc/target/intel_gen/pipeline.cc

```diff
@@ -16,43 +19,52 @@
 #include "pmlc/dialect/tile/transforms/passes.h"
 
 using namespace mlir; // NOLINT[build/namespaces]
-using pmlc::conversion::pxa_to_affine::createLowerPXAToAffinePass;
 
 namespace pmlc::target::intel_gen {
 
 namespace {
 
 void addToPipeline(OpPassManager &pm) {
-  pm.addPass(pmlc::dialect::tile::createComputeBoundsPass());
-  pm.addNestedPass<FuncOp>(createCanonicalizerPass());
-  pm.addNestedPass<FuncOp>(createCSEPass());
+  pm.addPass(dialect::tile::createComputeBoundsPass());
+  // pm.addPass(dialect::tile::createPadPass());
+  pm.addPass(createCanonicalizerPass());
+  pm.addPass(createCSEPass());
 
-  pm.addPass(pmlc::conversion::tile_to_pxa::createLowerTileToPXAPass());
-  pm.addNestedPass<FuncOp>(createCanonicalizerPass());
-  pm.addNestedPass<FuncOp>(createCSEPass());
+  pm.addPass(conversion::tile_to_pxa::createLowerTileToPXAPass());
+  pm.addPass(createCanonicalizerPass());
+  pm.addPass(createCSEPass());
 
   // TODO: do optimizations here
 
-  pm.addPass(createLowerPXAToAffinePass());
-  pm.addNestedPass<FuncOp>(createCanonicalizerPass());
-  pm.addNestedPass<FuncOp>(createCSEPass());
-
+  // pm.addPass(std::make_unique<UpdateWorkGroupSizePass>(workGroupSize));
+  // pm.addPass(std::make_unique<IREETileLinalgPass>());
+  // pm.addPass(createConvertLinalgToLoopsPass());
+  pm.addPass(conversion::pxa_to_affine::createLowerPXAToAffinePass());
   pm.addPass(createLowerAffinePass());
-  pm.addNestedPass<FuncOp>(createCanonicalizerPass());
-  pm.addNestedPass<FuncOp>(createCSEPass());
+  pm.addPass(createCanonicalizerPass());
+  pm.addPass(createCSEPass());
 
+  // pm.addPass(std::make_unique<LoopsToGPUPass>());
   pm.addPass(createSimpleLoopsToGPUPass(1, 1));
-  pm.addNestedPass<FuncOp>(createCanonicalizerPass());
-  pm.addNestedPass<FuncOp>(createCSEPass());
-
+  pm.addPass(createCanonicalizerPass());
   pm.addPass(createGpuKernelOutliningPass());
```

Comment:
@YangleiZouIntel This is the pass that will generate the host-side `main` function for us.